app/src/main/java/com/alexbar/layoutbasic/screen/confirm_payment/widgets/AddNewCreditCardDialog.kt

```diff
@@ -0,0 +1,169 @@
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(horizontal = dimen_16_dp),
+                        colors = TextFieldDefaults.colors(
+                            focusedContainerColor = Color.White,
+                            unfocusedContainerColor = Color.White
+                        ),
+                    )
+
+                    Row(
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(dimen_16_dp),
+                        horizontalArrangement = Arrangement.SpaceBetween
+                    ) {
+                        TextField(
+                            modifier = Modifier.weight(1f),
+                            value = expirationMonth,
+                            onValueChange = { expirationMonth = it },
+                            label = { Text(add_credit_card_month) },
+                            colors = TextFieldDefaults.colors(
+                                focusedContainerColor = Color.White,
+                                unfocusedContainerColor = Color.White
+                            ),
+                        )
+                        Spacer(Modifier.width(dimen_16_dp))
+                        TextField(
+                            modifier = Modifier.weight(1f),
+                            value = expirationYear,
+                            onValueChange = { expirationYear = it },
+                            label = { Text(add_credit_card_year) },
+                            colors = TextFieldDefaults.colors(
+                                focusedContainerColor = Color.White,
+                                unfocusedContainerColor = Color.White
+                            ),
+                        )
+                    }
+
+                    Row(
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(dimen_16_dp),
+                        horizontalArrangement = Arrangement.End,
+                    ) {
+                        TextButton(onClick = onDismissRequest) {
+                            Text(add_credit_card_cancel)
+                        }
+                        TextButton(onClick = {
+                            onCreateCard(
+                                CreditCardInfo(
+                                    name,
+                                    number,
+                                    expirationMonth,
+                                    expirationYear
+                                )
+                            )
+                            onDismissRequest()
+                        }) {
+                            Text(add_credit_card_save)
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+data class CreditCardInfo(
```

Comment:
One point here :
Please keep all your classes in a separate file
one of the solid principles is single functionality which means each class in our project should have one responsibility...